api/app/domain/analyzers/grasslands_analyzer.py

```diff
@@ -62,25 +60,20 @@ async def analyze(self, analysis: Analysis):
             )
             dfs = await self.compute_engine.gather(dd_df_futures)
             combined_results_df = await self.compute_engine.compute(dd.concat(dfs))
-
-            pass
+            results = combined_results_df.to_dict(orient="list")
 
         analyzed_analysis = Analysis(
-            combined_results_df.to_dict(orient="list"),
+            results,
             analysis.metadata,
             AnalysisStatus.saved,
         )
         await self.analysis_repository.store_analysis(
             grasslands_analytics_in.thumbprint(), analyzed_analysis
         )
 
-    @staticmethod
-    def analyze_admin_areas(
-        gadm_ids, parquet_file, start_year, end_year
-    ) -> pd.DataFrame:
-        query = f"select year, area_ha, aoi_id from '{parquet_file}' where aoi_id in {gadm_ids} and year >= {start_year} and year <= {end_year} order by aoi_id, year"
-        initialize_duckdb()
-        df = duckdb.query(query).df()
+    async def analyze_admin_areas(self, gadm_ids, start_year, end_year) -> DataFrame:
+        query = f"select year, area_ha, aoi_id from data_source where aoi_id in {gadm_ids} and year >= {start_year} and year <= {end_year} order by aoi_id, year"
+        df = await self.duckdb_query_service.execute(query)
         df["aoi_type"] = "admin"
```

Comment:
I think you need to make this a list based on @jterry64 's changes:
https://github.com/wri/project-zeno-data-infra/blob/7eafd50785438186b9c9c5cd37d88f31f6d8406b/api/app/domain/analyzers/land_cover_composition_analyzer.py#L95

Reply:
Done, thanks!